CONTRIBUTING.md

```diff
@@ -44,6 +44,8 @@ When you're ready, submit a pull request!
 
 * All changes must have adequate test coverage. For instruction on writing and running the various test suites, see Testing your changes.
 
+* If relevant, add a changelog entry.
```

Comment:
Shouldn't CHANGELOG entries be reserved for release preparation? The party issuing a release would be responsible for that, which should always be a maintainer (us) in the project. Like the commit that introduces the CHANGELOG entry should probably always come from the maintainer team right before the tag commit, no?

My concern with adding this here is that we may start seeing CHANGELOG commits in PRs that aren't applicable to an actual release cadence, and that will cause churn in our PR process because we'll ask them to remove those commits.

Reply:
Agree with @jon-whit .  I prefer having the changelog being modified by the maintainer when there is a release. In this way, it will be i) truly associated with the release and ii) less painful if we ever want to cherry-pick fix to associate with a release.

Reply:
+1 on both @jon-whit and @adriantam's points.

Though the commit message & description does to be clear:
1- for the team when reviewing the PR
2- to use for the merge commit that we will reference when creating the release notes